packages/plugin-ext/src/plugin/output-channel/logoutput-channel.ts

```diff
@@ -0,0 +1,80 @@
+// *****************************************************************************
+// Copyright (C) 2023 STMicroelectronics and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { AbstractMessageLogger, DEFAULT_LOG_LEVEL, LogLevel } from '@theia/monaco-editor-core/esm/vs/platform/log/common/log';
+import * as theia from '@theia/plugin';
+
+import { OutputChannelRegistryMain, PluginInfo } from '../../common/plugin-api-rpc';
+import { toLogLevel } from '../type-converters';
+
+export class LogOutputChannelImpl extends AbstractMessageLogger implements theia.LogOutputChannel {
+
+    private _disposed: boolean = false;
+    get disposed(): boolean { return this._disposed; }
+
+    override onDidChangeLogLevel: theia.Event<theia.LogLevel>;
+
+    constructor(readonly name: string, protected proxy: OutputChannelRegistryMain, protected readonly pluginInfo: PluginInfo) {
```

Comment:
is the `proxy` not readonly?

Reply:
Good catch, thank you, I agree and also think it should be readonly. However, it wasn't readonly in the `OutputChannelImpl` either, hence I updated it for both channels.